packages/react-keytips/src/hooks/useKeytipRef.ts

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { useEventService } from './useEventService';
+import { EVENTS } from '../constants';
+import { usePrevious } from '@fluentui/react-utilities';
+import type { KeytipProps } from '../components/Keytip';
+import { sequencesToID } from '../utilities';
+
+const isEqualArray = (a: string[], b: string[]) => {
+  return a.length === b.length && a.every((v, i) => v === b[i]);
+};
+
+export const useKeytipRef = <
+  T extends HTMLElement = HTMLButtonElement | HTMLAnchorElement
+>(
+  keytip: Omit<KeytipProps, 'uniqueId'>
+): React.Dispatch<React.SetStateAction<T | null>> => {
+  const uniqueId = React.useId();
+  const [node, setNode] = React.useState<T | null>(null);
+  const { dispatch } = useEventService();
+
+  const ktp = React.useMemo(
+    () => ({
+      ...keytip,
+      uniqueId,
+      id: sequencesToID(keytip.keySequences),
+      positioning: {
+        target: node,
+        ...keytip.positioning,
+      },
+    }),
+    [keytip, uniqueId, node]
+  );
+
+  const prevKeytip = usePrevious(ktp);
+
+  React.useEffect(() => {
+    if (prevKeytip) {
+      if (!isEqualArray(prevKeytip.keySequences, ktp.keySequences)) {
+        dispatch(EVENTS.KEYTIP_UPDATED, ktp);
+      }
+    }
+  });
```

Comment:
Effect dependencies are not specified

Reply:
this is on purpose, so this check runs every time ;) in v8 it was kinda similar [for update logic](https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/KeytipData/useKeytipData.ts#L29)

Reply:
Could you please add a note stating it's done on purpose? This way, we can avoid similar questions in future 😄